spec/lib/diaspora_federation/discovery/discovery_spec.rb

```diff
@@ -146,6 +116,84 @@ module DiasporaFederation
         expect { subject.fetch_and_save }.to raise_error Discovery::DiscoveryError
       end
 
+      context "http fallback" do
+        context "http fallback disabled (default)" do
+          it "falls back to legacy WebFinger if https fails with 404" do
+            stub_request(:get, "https://localhost:3000/.well-known/webfinger?resource=acct:#{account}")
+              .to_return(status: 404)
+            stub_request(:get, "https://localhost:3000/.well-known/host-meta")
+              .to_return(status: 200, body: host_meta_xrd)
+            stub_request(:get, "http://localhost:3000/.well-known/webfinger.xml?resource=acct:#{account}")
+              .to_return(status: 200, body: webfinger_xrd)
+            stub_request(:get, "http://localhost:3000/hcard/users/#{alice.guid}")
+              .to_return(status: 200, body: hcard_html)
+
+            expect_callback(:save_person_after_webfinger, kind_of(Entities::Person))
+            person = subject.fetch_and_save
+
+            expect(person.guid).to eq(alice.guid)
+            expect(person.diaspora_id).to eq(account)
+          end
+
+          it "falls back to legacy WebFinger if https fails with ssl error" do
+            stub_request(:get, "https://localhost:3000/.well-known/webfinger?resource=acct:#{account}")
+              .to_raise(OpenSSL::SSL::SSLError)
+            stub_request(:get, "https://localhost:3000/.well-known/host-meta")
+              .to_return(status: 200, body: host_meta_xrd)
```

Comment:
I did that now, but now this test is exactly the same as the test below for legacy WebFinger. But since it still tests what it should test it doesn't matter, and it will change once legacy WebFinger is removed.